pkg/controller/common/helm/patch.go

```diff
@@ -26,22 +25,30 @@ func NewPatchFactory(k8sClient client.Client) *PatchFactory {
 }
 
 // CreatePatch creates a patch based on the current and new versions of an object
-func (p *PatchFactory) CreatePatch(current, new runtime.Object) (Patch, error) {
+func (p *PatchFactory) CreatePatch(current, new *unstructured.Unstructured) (Patch, error) {
 	return &basicPatch{client: p.client, oldObj: current, newObj: new}, nil
 }
 
 type basicPatch struct {
 	client client.Client
-	oldObj runtime.Object
-	newObj runtime.Object
+	oldObj *unstructured.Unstructured
+	newObj *unstructured.Unstructured
 }
 
 func (p *basicPatch) Apply(ctx context.Context) (*unstructured.Unstructured, error) {
-	if err := p.client.Patch(ctx, p.newObj, client.Merge, client.FieldOwner("istio-operator")); err != nil {
-		return nil, err
+	var patch client.Patch
+	if originalBytes, err := util.GetOriginalConfiguration(p.oldObj); err == nil && len(originalBytes) > 0 {
```

Comment:
This part is a bit strange to me. The problem is that the lastAppliedConfiguration annotation is explicitly set by us, but it won't be updated when making changes using `kubectl` (at least not for me) - so we're creating a Patch based on possibly outdated information. I'm not sure that was intended here? It's neat, because it allows you to change a container image for example without the operator overwriting your change, but it will also mean that changes made by the user are never overwritten except the SMCP change triggering the patch is touching those very fields

Reply:
Yes, that was the intent, but I'm still parsing @luksa's comments below.